src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseReport.java

```diff
@@ -0,0 +1,603 @@
+  }
+
+  private List<String> labelListHeader(String header, List<SampleLabel> labelList) {
+    // needs to have one column for the time point, then two columns for each sampleLabel
+    // which are for the "Samples" and "Cases" columns in the {#samplesCasesHeader} method
+    List<String> list = new ArrayList<>();
+    list.add(header);
+    for (SampleLabel label : labelList) {
+      list.add(label.getKey());
+      list.add("");
+    }
+    return list;
+  }
+
+  private List<String> sampleCasesHeader(List<SampleLabel> labelList) {
+    // first column is "Site", then a pair of "Samples" + "Cases" columns for each sampleLabel in the list
+    List<String> sampleCasesList = new ArrayList<>();
+    sampleCasesList.add("Site");
+    for (int i = 0; i < labelList.size(); i++) {
+      sampleCasesList.add(NUM_SAMPLES);
+      sampleCasesList.add(NUM_CASES);
+    }
+    return sampleCasesList;
+  }
+
+  @Override
+  protected List<ColumnDefinition> getColumns() {
+    String[] row = makeBlankRow();
+    row[0] = String.format("Samples Received To Date (%s)", new SimpleDateFormat(DATE_FORMAT).format(new Date()));
+    List<ColumnDefinition> columns = new ArrayList<>();
+    for (int i = 0; i < row.length; i++) {
+      columns.add(new ColumnDefinition(row[i]));
+    }
+    return columns;
+  }
+
+  private final int columnCount = labelListHeader("", labelListInventory).size(); // inventory is the table with the most columns
+
+  @Override
+  protected int getRowCount() {
+    return 2 // headers
+        + toDateRandom.size()
+        + 4 // total, blank, headers
+        + toDateSix.size()
+        + 4 // total, blank, headers
+        + toDateTwelve.size()
+        + 4 // total, blank, headers
+        + toDateEighteen.size()
+        + 4 // total, blank, headers
+        + toDateTwentyFour.size()
+        + 4 // total, blank, headers
+        + toDateSlides.size()
+        + 5 // total, blank, this-month, headers
+        + thisMonthRandom.size()
+        + 4 // total, blank, headers
+        + thisMonthSix.size()
+        + 4 // total, blank, headers
+        + thisMonthTwelve.size()
+        + 4 // total, blank, headers
+        + thisMonthEighteen.size()
+        + 4 // total, blank, headers
+        + thisMonthTwentyFour.size()
+        + 4 // total, blank, headers
+        + thisMonthSlides.size()
+        + 5 // total, blank, inventory, headers
+        + inventoryAvailable.size()
+        + 4 // inventory total plus extra rows between headings (To Date, This Month, Inventory)
+        + 2 // blank, header
+        + barcodeAndNameCodingMismatch.size()
+        + 2 // blank, header
+        + slidesWithoutSlideTimepoint.size()
+    ;
+  }
+
+  @Override
+  protected String[] getRow(int rowNum) {
+    String[] row = null;
+    row = maybeInList(TimePoint.RANDOMIZATION.getKey(), labelListLong, toDateRandom, rowNum);
+    if (row.length != 0) return row;
+    rowNum -= (toDateRandom.size() + 4); // list + (blank, 2 headers, total)
```

Comment:
**Would probably be simpler to create another implementation of `Report` rather than using `TableReport` for all of these reports that are not simple tables. A lot of weirdness could be avoided

Reply:
Yes, I'll look at doing this. Will update soon!

Reply:
I've created a [ticket](https://jira.oicr.on.ca/browse/GR-666) for adding another implementation. Ok to merge this for now and refactor later?

Reply:
Yes, I tried to indicate that most of my comments on this PR were non-essential